scripts/simplify_network.py

```diff
@@ -738,6 +741,118 @@ def drop_isolated_nodes(n, threshold):
     return n
 
 
+def merge_into_network(n, aggregation_strategies=dict()):
+    """
+    Find isolated nodes in the network and merge those of them which have load
+    value below than a specified threshold into a single isolated node which
+    represents all the remote generation.
+
+    Parameters
+    ----------
+    n : PyPSA.Network
+        Original network
+    threshold : float
+        Load power used as a threshold to merge isolated nodes
+
+    Returns
+    -------
+    modified network
+    """
+    # keep original values of the overall load and generation in the network
+    # to track changes due to drop of buses
+    generators_mean_origin = n.generators.p_nom.mean()
+    load_mean_origin = n.loads_t.p_set.mean().mean()
+
+    n.determine_network_topology()
+
+    # duplicated sub-networks mean that there is at least one interconnection between buses
+    i_islands = n.buses[
+        (~n.buses.duplicated(subset=["sub_network"], keep=False))
+        & (n.buses.carrier == "AC")
+    ].index
+
+    # TODO filtering may be applied to decide if the isolated buses should be fetched
+    # # isolated buses with load below than a specified threshold should be merged
+    # i_load_islands = n.loads_t.p_set.columns.intersection(i_islands)
+    # i_suffic_load = i_load_islands[
+    #     n.loads_t.p_set[i_load_islands].mean(axis=0) <= threshold
+    # ]
+
+    # return the original network if no isolated nodes are detected
+    if len(i_islands) == 0:
+        return n, n.buses.index.to_series()
+
+    i_connected = n.buses.loc[n.buses.carrier == "AC"].index.difference(i_islands)
+
+    points_buses = np.array(
+        list(zip(n.buses.loc[i_connected].x, n.buses.loc[i_connected].y))
+    )
+    islands_points = np.array(
+        list(zip(n.buses.loc[i_islands].x, n.buses.loc[i_islands].y))
+    )
+
+    gds_buses = gpd.GeoSeries(map(Point, points_buses))
+    gds_islands = gpd.GeoSeries(map(Point, islands_points))
+
+    gdf_buses = gpd.GeoDataFrame(geometry=gds_buses)
+    gdf_islands = gpd.GeoDataFrame(geometry=gds_islands)
```

Comment:
to create a geodatafarme (let's say `gdf_buses`)of the entire buses df we may use this function, that should be quite compact

Reply:
Agree that it could be implemented in a more elegant way :) more  Wrapped-up into a function transformation into a geopandas dataframe.

Have I understood your idea correctly? :)